src/Files.App/Extensions/MessageFormatExtensions.cs

```diff
@@ -15,10 +14,19 @@ public static class MessageFormatExtensions
 		private static readonly ResourceMap resourcesTree = new ResourceManager().MainResourceMap.TryGetSubtree("Resources");
 
 		// CultureInfo based on the application's primary language override
-		private static readonly CultureInfo locale = new(ApplicationLanguages.PrimaryLanguageOverride);
+		private static readonly CultureInfo locale = default!;
 
 		// Message formatter with caching enabled, using the current UI culture's two-letter ISO language name
-		private static readonly MessageFormatter formatter = new(useCache: true, locale: locale.TwoLetterISOLanguageName);
+		private static readonly MessageFormatter formatter = default!;
+
+		static MessageFormatExtensions()
+		{
+			if (locale is not null)
+				return;
+
+			locale = new(AppLanguageHelper.SelectedLanguage);
+			formatter = new(useCache: true, locale: locale.TwoLetterISOLanguageName);
+		}
```

Comment:
Can't we simply replace `new(ApplicationLanguages.PrimaryLanguageOverride)` with `new(AppLanguageHelper.SelectedLanguage)`?